### DISCUSSION TITLE
[2024-01-15] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by vitvakatu** [enso-org/enso#8763] [Weekly Demo Day]
@vitvakatu 

- #8758 
- https://github.com/enso-org/enso/issues/8699

https://github.com/enso-org/enso/assets/6566674/1300c8e4-5520-4671-9bb1-2d0acebd26d3

@farmaazon 

- https://github.com/enso-org/enso/issues/8683
- https://github.com/enso-org/enso/issues/8727
- https://github.com/enso-org/enso/pull/8749

[Screencast from 2024-01-11 15-14-08.webm](https://github.com/enso-org/enso/assets/3919101/afdf2c1a-3110-4ddd-83ff-fd0c3551afb6)

**Comment by MichaelMauderer** [comment]
* Working on debugging #8604
  * Resolved some initial issues which restored functionality .. sometimes. 
  * Appears to be some more in-depth issue with the AST manipulation API that still needs more investigation. 

[Peek 2024-01-15 15-19.webm](https://github.com/enso-org/enso/assets/1428930/d24b464e-626e-483c-bc5e-4a1501f37053)

**Comment by kazcw** [comment]
- #8683

Working on new synchronization #8237.

**Comment by mwu-tow** [comment]
Nightly build issues:
* Accepting updated agreements on company's Apple Developer page.
* Follow up fix to the #8587 — #8757

Other CI improvements:
* Remove old GUI builds from PR checks. #8721
* Adding sanity test for the native Engine Runner. #8724
* Not building native runner twice on PR check. #8726

Soon:
* Cleaning old benchmark reports on CI. #8762

**Comment by Frizi** [comment]
- Implemented reordering nodes in code after edge connection #8494 
- Improved placement of argument placeholders, fixed many cases of placeholders being temporarily unstable after node edit.

**Comment by hubertp** [comment]
Relevant backend changes:
- Return function schema for atom constructors #8743 
- Generate UUIDs on demand #8728
